src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -351,8 +356,37 @@ public static bool TryText(FormattingInfo formatInfo, IRContext irContext, Formu
                 return false;
             }
 
-            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
-
+            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
+
+            bool hasFormatCulture = false;
+            string enUsCultureName = "en-US";
+            CultureInfo formatCulture = culture;
+            string revFormatStr = formatString;
+
+            if (!string.IsNullOrEmpty(formatString) && formatString.StartsWith("[$-", StringComparison.OrdinalIgnoreCase))
+            {
+                int startIdx = formatString.IndexOf("[$-", StringComparison.Ordinal) + 3;
+                int endIdx = formatString.IndexOf("]", StringComparison.Ordinal);
+
+                if (endIdx > 0)
+                {
+                    hasFormatCulture = true;
+                    string formatCultureName = formatString.Substring(startIdx, endIdx - startIdx);
+
+                    if (!TryGetCulture(formatCultureName, out formatCulture))
+                    {
+                        result = new StringValue(irContext, formatCultureName);
+                        return false;
+                    }
+
+                    formatString = formatString.Substring(endIdx + 1);
+                    revFormatStr = formatString.Replace(",", "!");
+                    revFormatStr = revFormatStr.Replace(".", ",");
+                    revFormatStr = revFormatStr.Replace(" ", ",");
+                    revFormatStr = revFormatStr.Replace("!", ".");

```

Comment:
Can we pull the separators from the formatCulture itself?

Reply:
In this one I tried to find the reverse version of input format string. So I cannot get separators by formatculture.

Reply:
Can you do something like the code below?

    var nfi = formatCulture.NumberFormat;
    formatString = formatString.Substring(endIdx + 1);
    revFormatStr = formatString.Replace(nfi.NumberDecimalSeparator, "\uFEFF");
    revFormatStr = revFormatStr.Replace(nfi.NumberGroupSeparator, ",");
    revFormatStr = revFormatStr.Replace("\uFEFF", ".");

This way we can get the appropriate separator for the locale. Even languages that have "space" as the thousands grouping have different spaces: fr-FR has \u202F (narrow no-break space), hu-HU (and many others) has \u00A0 (no-break space)

Reply:
Thanks for sharing. Update the code. It is ood that this revFormatStr = formatString.Replace(numberCultureFormat.NumberGroupSeparator, "\uFEFF"); does not work when NumberGroupSeparator is space. So I have to add if line to update for space case.